bulk-update-github/bulk-update-github.py

```diff
@@ -0,0 +1,59 @@
+import os
+import sys
+from git import Repo
+from github import Github
+
+if len(sys.argv) < 7:
+    print("6 arguments are needed. Exiting...")
+    sys.exit(1)
+
+github_org = sys.argv[1]
+github_tkn = sys.argv[2]
+dir_lookup = sys.argv[3]
+string_old = sys.argv[4]
+string_new = sys.argv[5]
+msg_commit = sys.argv[6]
+
+dir_repos = os.getcwd() + '/repos'
+os.mkdir(dir_repos)
+repos = Github(github_tkn).get_organization(github_org).get_repos()
+print("\n" + str(repos.totalCount) + " repos in total.")
+
+for repo in repos:
+    print("\nRepo '" + repo.name + "'...")
+    if not repo.archived:
+        os.chdir(dir_repos)
+        repo_url_withtoken = repo.clone_url.replace('https://', 'https://' + github_tkn + '@')
+        repo_cloned = Repo.clone_from(repo_url_withtoken, repo.name, depth=1)
+        os.chdir(repo.name)
+
+        if (os.path.exists(dir_lookup)):
+            files_updated = 0
+
+            for file_name in os.listdir(dir_lookup):
+                file = os.path.join(dir_lookup, file_name)
+
+                with open(file, 'r') as f:
+                    content = f.read()
+
+                new_content = content.replace(string_old, string_new)
+
+                with open(file, 'w') as f:
+                    if new_content != content:
+                        files_updated += 1
+                        f.write(new_content)
+
+            if files_updated > 0:
+                print(str(files_updated) + " files with changes. Updating.")
+                repo_cloned.index.add(dir_lookup)
+                repo_cloned.index.commit(msg_commit)
+                repo_cloned.remotes.origin.push()
+
+            else:
+                print("No files with changes. Skipping.")
+        else:
+            print("No directory to lookup. Skipping.")
+    else:
+        print("Archived repo. Skipping.")
+
+print("\nAll done.")
```

Comment:
In python, you'll want to put your entrypoint logic under an `if __name__ == '__main__':` block.

It is not strictly necessary, but it is good practice and as a bonus, I find that it helps with legibility.